app/javascript/components/stripe/StripeForm.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useCallback, useState } from 'react'
+import { Icon } from '../common'
+import { sendRequest } from '../../utils/send-request'
+import { loadStripe, StripeCardElementChangeEvent } from '@stripe/stripe-js'
+import {
+  Elements,
+  CardElement,
+  useStripe,
+  useElements,
+} from '@stripe/react-stripe-js'
+import { fetchJSON } from '../../utils/fetch-json'
+
+type PaymentIntent = {
+  id: string
+  clientSecret: string
+}
+export type PaymentIntentType = 'payment' | 'subscription'
+
+// Make sure to call `loadStripe` outside of a component’s render to avoid
+// recreating the `Stripe` object on every render.
+const stripePromise = loadStripe(
+  'pk_test_51IDGMXEoOT0Jqx0UcoKlkvB7O0VDvFdCBvOCiWiKv6CkSnkZn7IG6cIHuCWg7cegGogYJSy8WsaKzwFHQqN75T7b00d56MtilB'
```

Comment:
We can probably put this into a meta tag so that we can store the secret key inside Rails.

Reply:
Agreed